<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <param name="robot_description"
    textfile="$(find sensor_driver)/urdf/multivelodyne_3d.urdf" />
    
  <node name="robot_state_publisher" pkg="robot_state_publisher"
    type="robot_state_publisher" />
    
  <arg name="rviz" default="false" />


  <arg name="configfile" default="$(find sensor_driver)/config/configmultilidar-cwproject.xml" />
  <arg name="logdir" default="--log_dir=/home/zhubc/lab/code/20181108/catkin_ws/log" />
  <arg name="glogparam" default="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=5   $(arg logdir)" />

  <node pkg="sensor_driver" type="masternode" name="masternode" args="$(arg glogparam)" output="screen">
  	<param name="config_file" value="$(arg configfile)"/>
  </node>



  <node pkg="colorize_pointcloud" type="colorize_pointcloud_node" name="colorize_pointcloud_node" args="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=2   $(arg logdir)" output="screen">
    <param name="pointcloud_topic" value="/lidar_cloud_calibrated"/>
    <param name="front_camera_on" value="true"/>
    <rosparam param="front_x_range">[-7,7]</rosparam>
    <rosparam param="front_y_range">[3,30]</rosparam>
    <rosparam param="front_z_range">[-1,4]</rosparam>
    <rosparam param="left_x_range">[-30,-3]</rosparam>
    <rosparam param="left_y_range">[-10,10]</rosparam>
    <rosparam param="left_z_range">[-1,4]</rosparam>
    <rosparam param="right_x_range">[3,30]</rosparam>
    <rosparam param="right_y_range">[-10,10]</rosparam>
    <rosparam param="right_z_range">[-1,4]</rosparam>
    <param name="front_camera_topic" value="/rgb_front/image_raw/compressed"/>
    <param name="front_camera_lidar_extrinsic_path" value="$(find colorize_pointcloud)/extrinsics/extrinsics_rgb_front_and_RS32.yaml"/>
    <param name="left_camera_on" value="true"/>
    <param name="left_camera_topic" value="/rgb_left/image_raw/compressed"/>
    <param name="left_camera_lidar_extrinsic_path" value="$(find colorize_pointcloud)/extrinsics/extrinsics_rgb_left_and_RS32.yaml"/>
    <param name="right_camera_on" value="true"/>
    <param name="right_camera_topic" value="/rgb_right/image_raw/compressed"/>
    <param name="right_camera_lidar_extrinsic_path" value="$(find colorize_pointcloud)/extrinsics/extrinsics_rgb_right_and_RS32.yaml"/>
    <param name="display_on" value="true"/>
  </node>


 <node pkg="sensor_driver" type="getmultirslidardata" name="getmultirslidardata" args="--logtostderr=0  --colorlogtostderr=1 --stderrthreshold=2   $(arg logdir)" output="screen">
  <param name="fix_frame" value="true"/>
  </node>
 

  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find obstacle_detection)/rviz_cfg/covgrid_mapping.rviz" />
  </group>

</launch>
